include/contraction/linearContraction.hpp

```diff
@@ -152,46 +143,9 @@ class Pgr_linear {
         }
     }
 
-
-
-    /**
-     *
-     * u ----e1{v1}----> v ----e2{v2}----> w
-     *
-     * e1: min cost edge from u to v
-     * e2: min cost edge from v to w
-     *
-     * result:
-     * u ---{v+v1+v2}---> w
-     *
-     */
-    void process_shortcut(G &graph, V u, V v, V w) {
-        auto e1 = graph.get_min_cost_edge(u, v);
-        auto e2 = graph.get_min_cost_edge(v, w);
-
-        if (std::get<2>(e1) && std::get<2>(e2)) {
-            auto contracted_vertices = std::get<1>(e1) + std::get<1>(e2);
-            double cost = std::get<0>(e1) + std::get<0>(e2);
-            contracted_vertices += graph[v].id;
-            contracted_vertices += graph[v].contracted_vertices();
-
-            // Create shortcut
-            CH_edge shortcut(
-                    get_next_id(),
-                    graph[u].id,
-                    graph[w].id,
-                    cost);
-            shortcut.contracted_vertices() = contracted_vertices;
-
-            graph.add_shortcut(shortcut, u, w);
-        }
-    }
-
-
 private:
     Identifiers<V> m_linearVertices;
     Identifiers<V> m_forbiddenVertices;
-
     int64_t last_edge_id;
```

Comment:
Do you still need this attribute?

Reply:
Yes, it is still used to number the new edges with negative numbers, it gives the last one created.

Reply:
Hum, when scanning the code base on your branch, I see that you initialize it to 0, you do not decrease it any more (the `get_next_id` method is removed in the branch). Plus, you do not actually *exploit* this attributes...

Reply:
The commit in your first link does not exist on Github.

Then if it becomes required in #7, why don't you set it in #7?

> And it is exploited, in two of the contraction methods to number the output shortcut edges from [here](https://github.com/AuBousquet/pgrouting/pull/7/files#diff-6e17355b753c856617876264d72e350d1663de0397915e50b7a85de51876cec4R385).

we do not necessarily have to replicate *all* the attributes in every child class, don't we? (It does not seem like a problem to have a specific behavior in one of the child class, i.e. that's why we need this specific child class.)

Reply:
no, but it is still used by the linear contraction method, even in this PR

Reply:
On the last version of the branch:

```bash
$ git grep last_edge_id
include/contraction/linearContraction.hpp:    Pgr_linear():last_edge_id(0) {}
include/contraction/linearContraction.hpp:    int64_t last_edge_id;
```

I honestly don't understand where it is used. :/

Reply:
Understood... sorry, it is replicated from the mother class and becomes now useless in the child class.